### PR TITLE
fix(validations): fix schema validations

### PR DESCRIPTION
**Pull request by gperdomor**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #124 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: lib/config.module.ts [hunk 1]
```diff
@@ -38,39 +38,40 @@ export class ConfigModule {
...
+      validatedEnvConfig = validatedConfig;
+      this.assignVariablesToProcess(validatedConfig);
+    } else {
+      this.assignVariablesToProcess(config);
     }
+
     const isConfigToLoad = options.load && options.load.length;
     const providers = (options.load || [])
-      .map(factory =>
+      .map((factory) =>
```
Comment (prateekkathal) [inline lib/config.module.ts:69]: I'd remove the parenthesis added by your IDE everywhere. This change is unnecessary and will cause inconsistencies.

**Comment by prateekkathal** [comment]
Thanks for cleaning the code up. I feel the changes make sense.... Can't think of any cases that might break or get missed because of this 👍 

Whenever @kamilmysliwiec gets time to look at this, he'll leave his comments or just merge it. 😄 

Thanks again!

File: lib/config.module.ts [hunk 2]
```diff
@@ -38,39 +38,40 @@ export class ConfigModule {
...
-        if (error) {
-          throw new Error(`Config validation error: ${error.message}`);
-        }
-        validatedEnvConfig = validatedConfig;
-        this.assignVariablesToProcess(validatedConfig);
-      } else {
-        const config = this.loadEnvFile(options);
-        this.assignVariablesToProcess(config);
+
+    let config = this.loadEnvFile(options);
```
Comment (kamilmysliwiec) [inline lib/config.module.ts:42]: Why are we calling the `loadEnvFile()` method even if `options. ignoreEnvVars` is set to `true`?

Comment (kamilmysliwiec) [reply lib/config.module.ts:42]: I can see the condition in the `loadEnvFile()` method now. Anyways, wouldn't it be cleaner to do something like this:

```suggestion
    let config = options.ignoreEnvVars ? {} : this.loadEnvFile(options);
```

instead? + removing the condition from the `loadEnvFile`

**Pull request merged**